### PR TITLE
feat: add leagues notifier for areas, relics, and tasks

### LINKED ISSUE #356: Include league messages in a separate notifier type
### Checklist

- [X] I've searched the [issues and pull requests](https://github.com/pajlads/Dinkplugin/issues?q=) for similar looking suggestions.
- [X] I've checked the `Unreleased` section of the [changelog](https://github.com/pajlads/DinkPlugin/blob/master/CHANGELOG.md) for newly added features that sound like my suggestion.

### Describe your Suggestion

Could we potentially have a league-specific task completion notification, separate (in a new notifier/category) from the rest of the existing notifications?

Could we also confirm that there is an “account type” label of something league specific to organise notifications as desired?

I don’t think anyone has the exact syntax yet but this is from last trailblazer:
IMG_4774

### Reasoning

_No response_

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #356

Notifies on:
* Area unlocked

image

&nbsp;

* Relic unlocked

image

&nbsp;

* Task completed

image

image

### CODE REVIEW

**Comment by Felanbird** [comment]
tasks work
task filtering works

File: src/main/java/dinkplugin/notifiers/LeaguesNotifier.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        thresholds.put(2_500, "Bronze");
+        thresholds.put(5_000, "Iron");
+        thresholds.put(10_000, "Steel");
+        thresholds.put(18_000, "Mithril");
+        thresholds.put(28_000, "Adamant");
+        thresholds.put(42_000, "Rune");
+        thresholds.put(56_000, "Dragon");
+        TROPHY_BY_POINTS = Collections.unmodifiableNavigableMap(thresholds);
+
+        TIER_BY_RELIC = Map.ofEntries(
```
Comment (iProdigy) [inline src/main/java/dinkplugin/notifiers/LeaguesNotifier.java:265]: In a future PR, can switch hardcoded relic names to dynamic lookup from structs/enums

**Review by Felanbird — APPROVED** [review]
I tested everything except region unlock, I believe it all works

**Comment by iProdigy** [comment]
I'll try to get a region unlock tonight before falling asleep to test that branch

**Comment by iProdigy** [comment]
area notif works

image

File: src/main/java/dinkplugin/notifiers/data/LeaguesAreaNotificationData.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class LeaguesAreaNotificationData extends NotificationData {
+    String area;
+    Integer index;
+    int tasksCompleted;
+    Integer tasksUntilNextArea;
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/data/LeaguesAreaNotificationData.java:16]: Mark as nullable

File: src/main/java/dinkplugin/notifiers/LeaguesNotifier.java [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+                notifyRelicUnlock(relic);
+            }
+        } else if (config.leaguesTaskCompletion()) {
+            LeagueTaskDifficulty tier;
+            Matcher matcher = TASK_REGEX.matcher(message);
+            if (matcher.find() &&
+                (tier = LeagueTaskDifficulty.TIER_BY_LOWER_NAME.get(matcher.group("tier"))) != null &&
+                tier.ordinal() >= config.leaguesTaskMinTier().ordinal()) {
+                notifyTaskCompletion(tier, matcher.group("task"));
+            }
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/LeaguesNotifier.java:105]: Could the tier-logic be scoped so the assignment doesn't happen inside the if-statement?
e.g.
```java
if (matcher.find()) {
  tier = ...;
  if (tier != null & tier.ordinal...) {
    notifyTaskCompletion
  }
}
```

**Review by pajlada — CHANGES_REQUESTED** [review]
Code looks good to me, two nitpicks is all I could find

**Review by pajlada — APPROVED** [review]

**Pull request merged**